migrator/migrations/policymigrationhelper/postgres_policy_migrator.go

```diff
@@ -0,0 +1,92 @@
+package policymigrationhelper
+
+import (
+	"context"
+	"embed"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	pglog "github.com/stackrox/rox/migrator/log"
+	"github.com/stackrox/rox/pkg/sac"
+)
+
+// MigratePoliciesWithDiffsAndStore migrates policies with the given diffs.
+// The policyDiffFS should be an embedded FS that satisfies the following conditions:
+// 1. It must contain a top-level directory called "policies_before_and_after".
+// 2. That directory must contain two subdirectories: "before" and "after".
+// 3. For each policy being migrated, there must be one copy in the "before" directory and one in the "after" directory.
+// 4. The file names for a policy should match the PolicyFileName in the corresponding PolicyDiff passed in the third argument.
+// This function then automatically computes the diff for each policy, and executes the migration.
+// This method requires the caller to provide funcs that check if a policy exists,
+// fetches policy from the store and will upsert a policy to the store
+func MigratePoliciesWithDiffsAndStore(policyDiffFS embed.FS,
```

Comment:
Why not take a pointer to the store interface directly?

Reply:
I did consider it, but I decided to do this for a few reasons:

1. More flexibility - if I wanted to instrument or put in a transaction or something I can always pass in a func that wraps the store method
2. This is agnostic to the store interface changing. if the interface were to change, then we have to go back and edit all migrations that used this method to have their "frozen" store interfaces implement the new method. I think touching existing migrations are risker so this allows us to remain decoupled.
3. Currently we create "frozen" store interfaces for every migration by copying the existing one and removing methods that are irrelevant. We can no longer remove anything (or at least have to stub it out) if we use the interface.

Reply:
The path we've chosen means func signatures on the store and hence on the migrator helper methods can't change either. So either way there are downsides.